client/cmd/dexc/config.go

```diff
@@ -141,11 +142,17 @@ func configure() (*Config, error) {
 		os.Exit(0)
 	}
 
-	// If the app directory has been changed, but the config file path hasn't,
-	// reform the config file path with the new directory.
-	if preCfg.AppData != defaultApplicationDirectory && preCfg.Config == defaultConfigPath {
-		preCfg.Config = filepath.Join(preCfg.AppData, configFilename)
+	// If the app directory has been changed, replace shortcut chars such
+	// as "~" with the full path.
+	if preCfg.AppData != defaultApplicationDirectory {
+		preCfg.AppData = core.CleanAndExpandPath(preCfg.AppData)
+		// If the app directory has been changed, but the config file path hasn't,
+		// reform the config file path with the new directory.
+		if preCfg.Config == defaultConfigPath {
+			preCfg.Config = filepath.Join(preCfg.AppData, configFilename)
+		}
 	}
```

Comment:
I apologize for this unrelated change. I often attempt to use home or `~` to with dexc and alway end up creating a floder named `~` in the current directory which is not intended. Hopefully this is ok to squeeze in here.